src/main/java/io/debezium/connector/spanner/kafka/internal/TaskSyncPublisher.java

```diff
@@ -91,7 +91,6 @@ private void publishSyncEvent(TaskSyncEvent taskSyncEvent) {
             lastTime = Instant.now();
         }
         catch (ExecutionException e) {
-            close();
```

Comment:
How will Buffered publisher and this thread terminate if close is not called?

Reply:
Hi Shantanu, if there is an ExecutionException here, the error handler will have accepted this exception.

The error handler sets the exception to true the ChangeEventQueue by calling setProducerThrowable: https://github.com/debezium/debezium-connector-spanner/blob/e61c7446f4e79ba6a2432064968964f12419eb71/src/main/java/io/debezium/connector/spanner/task/SynchronizationTaskContext.java#L230 

If the queue's producer exception is present, we will throw an exception when calling poll(): https://github.com/debezium/debezium/blob/0585c6beb81cc27e00bd7c9ce47f61bcbc8c9b00/debezium-core/src/main/java/io/debezium/connector/base/ChangeEventQueue.java#L306